File: leetcode/go/275.h-index-ii.go
func hIndex(citations []int) int {
    n, l, r, m := len(citations), 0, len(citations), 0
    for l < r {
        m = (l + r)/2
        if citations[m] >= n - m {
            r = m
        } else {
            l = m+1
        }
    }
    return n - l
}

File: leetcode/cpp/275.h-index-ii.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n=citations.size(), l=0, r=citations.size(), m=0;
        while (l < r) {
            m = (l + r)/2;
            if (citations[m] >= n - m) {
                r = m;
            } else {
                l = m+1;
            }
        }
        return n-l;
    }
};

